### PR TITLE
Added validator to verify scan config name is dns compatible

### PR DESCRIPTION
**Pull request by ksurabhi91**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by SimonBaeumer — APPROVED** [review]

**Review by dashrews78 — CHANGES_REQUESTED** [review]
I looked at the unit test failure.  This library validates a domain name, not a DNS name.  Basically it validates everything after the `@` in an email.  It may be just as simple in the short term to use a simple regex that implements the rules the UI describes on the create schedule page.
```
Name can contain only lowercase alphanumeric characters, '-' or '.', and start and end with an alphanumeric character.
```

File: central/complianceoperator/v2/scanconfigurations/service/service_impl.go [hunk 1]
```diff
@@ -74,6 +77,12 @@ func (s *serviceImpl) CreateComplianceScanConfiguration(ctx context.Context, req
 		return nil, errors.Wrap(errox.InvalidArgs, "Scan configuration name is required")
 	}
 
+	validName := domainRegexp.MatchString(req.GetScanName())
+
+	if !validName {
+		return nil, errors.Wrap(errox.InvalidArgs, "Scan configuration name is not comptaible with DNS name")
```
Comment (dashrews78) [inline central/complianceoperator/v2/scanconfigurations/service/service_impl.go:83]:
```suggestion
		return nil, errors.Wrapf(errox.InvalidArgs, "Scan configuration name %q is not comptaible with DNS name", req.GetScanName())
```

File: central/complianceoperator/v2/scanconfigurations/service/service_impl.go [hunk 2]
```diff
@@ -37,6 +38,8 @@ var (
 			"/v2.ComplianceScanConfigurationService/UpdateComplianceScanConfiguration",
 		},
 	})
+
+	domainRegexp = regexp.MustCompile(`^[a-z0-9][a-z0-9.-]*[a-z0-9]?$`)
```
Comment (dashrews78) [inline central/complianceoperator/v2/scanconfigurations/service/service_impl.go:42]:
```suggestion
	configNameRegexp = regexp.MustCompile(`^[a-z0-9][a-z0-9.-]*[a-z0-9]?$`)
```

File: central/complianceoperator/v2/scanconfigurations/service/service_impl.go [hunk 3]
```diff
@@ -74,6 +77,12 @@ func (s *serviceImpl) CreateComplianceScanConfiguration(ctx context.Context, req
 		return nil, errors.Wrap(errox.InvalidArgs, "Scan configuration name is required")
 	}
 
+	validName := domainRegexp.MatchString(req.GetScanName())
```
Comment (dashrews78) [inline central/complianceoperator/v2/scanconfigurations/service/service_impl.go:80]:
```suggestion
	validName := configNameRegexp.MatchString(req.GetScanName())
```

**Review by dashrews78 — APPROVED** [review]
Left a couple of nit suggestions I trust you to resolve

**Pull request merged**